scripts/build.py

```diff
@@ -0,0 +1,243 @@
+  collect_output (bool): Get output from the command. The string returned from this function
+       call will have the output from command execution.
+ Raises:
+  (subprocess.CalledProcessError): If command errored out.
+ Returns:
+  (str|None): If collect_output is provided, a string is returned
+     None if output is not requested.
+ """
+ if as_root_user and (is_mac_os() or is_linux_os()):
+  cmd.insert(0, 'sudo')
+
+ cmd_string = ' '.join(cmd)
+ print('Running cmd: {0}\n'.format(cmd_string))
+ if collect_output:
+  output = subprocess.check_output(cmd, cwd=cwd)
+  return output.decode('utf-8').strip()
+ else:
+  subprocess.call(cmd, cwd=cwd)
+
+
+def is_windows_os():
+ return platform.system() == 'Windows'
+
+
+def is_mac_os():
+ return platform.system() == 'Darwin'
+
+
+def is_linux_os():
+ return platform.system() == 'Linux'
+
+
+def get_vcpkg_triplet(arch):
+  """ Get vcpkg target triplet (platform definition).
+
+  Args:
+    arch (str): Architecture (eg: 'x86', 'x64').
+
+  Raises:
+    ValueError: If current OS is not win,mac or linux.
+  
+  Returns:
+    (str): Triplet name.
+       Eg: "x64-windows-static".
+  """
+  triplet_name = [arch]
+  if is_windows_os():
+    triplet_name.append('windows')
+    # For windows, default is to build dynamic. Hence we specify static.
+    # For mac/linux, default is to build static libraries
+    triplet_name.append('static')
+  elif is_mac_os():
+    triplet_name.append('osx')
+  elif is_linux_os():
+    triplet_name.append('linux')
+
+  triplet_name = '-'.join(triplet_name)
+  print("Using vcpkg triplet: {0}".format(triplet_name))
+  return triplet_name
+
+
+def get_vcpkg_executable_file_path():
+  """Get absolute path to vcpkg executable."""
+  vcpkg_root_dir = os.path.join(os.getcwd(), 'external', 'vcpkg')
+  if is_windows_os():
+    vcpkg_executable_file_path = os.path.join(vcpkg_root_dir, 'vcpkg.exe')
+  elif is_linux_os() or is_mac_os():
+    vcpkg_executable_file_path = os.path.join(vcpkg_root_dir, 'vcpkg')
+  return vcpkg_executable_file_path
+
+
+def get_vcpkg_installation_script_path():
+  """Get absolute path to the script used to build and install vcpkg."""
+  vcpkg_root_dir = os.path.join(os.getcwd(), 'external', 'vcpkg')
+  if is_windows_os():
+    script_absolute_path = os.path.join(vcpkg_root_dir, 'bootstrap-vcpkg.bat')
+  elif is_linux_os() or is_mac_os():
+    script_absolute_path = os.path.join(vcpkg_root_dir, 'bootstrap-vcpkg.sh')
+
+  return script_absolute_path
+
+
+
+def install_cpp_dependencies_with_vcpkg(arch):
+  """Install packages with vcpkg.
+
+  This does the following,
+    - initialize/update the vcpkg git submodule
+    - build vcpkg executable
+    - install packages via vcpkg.
+  Args:
+    arch (str): Architecture (eg: 'x86', 'x64').
+  """
+  # vcpkg is a submodule of the main cpp repo
+  # Ensure that the submodule is initialized and updated
+  run_command(['git', 'submodule', 'init'])
+  run_command(['git', 'submodule', 'update'])
```

Comment:
I'd be fine seeing this in main() since ultimately we may end up with other submodules.